dpnp/dpnp_iface_arraycreation.py

```diff
@@ -73,48 +73,64 @@ def arange(*args, **kwargs):
 
     Parameters
     ----------
-        start: Start of the interval.
-        stop: End of the interval.
-        step: Step width between each pair of consecutive values.
-        dtype: Data type specifier. It is inferred from other arguments by
-            default.
+    start : number, optional
+        Start of the interval.
+    stop : number
+        End of the interval.
+    step : number, optional
+        Step width between each pair of consecutive values.
+    dtype : dtype
+        Data type specifier. It is inferred from other arguments by default.
 
     Returns
     -------
-        inumpy.dparray: The 1-D array of range values.
+    arange : :obj:`dpnp.ndarray`
+        The 1-D array of range values.
+
+    Limitations
+    -----------
+    Parameter ``start`` is supported as integer only.
+    Parameters ``stop`` and ``step`` are supported as either integer or `None`.
+    Otherwise the function will be executed sequentially on CPU.
+
+    See Also
+    --------
+    :obj:`numpy.arange` : Return evenly spaced values within a given interval.
+    :obj:`dpnp.linspace` : Evenly spaced numbers with careful handling of endpoints.
+
+    Examples
+    --------
 
-    .. seealso:: :obj:`numpy.arange`
+    >>> import dpnp as np
+    >>> [i for i in np.arange(3)]
+    [0, 1, 2]
+    >>> [i for i in np.arange(3, 7)]
+    [3, 4, 5, 6]
+    >>> [i for i in np.arange(3, 7, 2)]
+    [3, 5]
 
     """
+    if use_origin_backend():
+        if not isinstance(start, int):
+            pass
+        if not isinstance(stop, int) or stop is not None:
+            pass
+        if not isinstance(step, int) or step is not None:
+            pass
+        else:
+            if dtype is None:
+                dtype = numpy.float64
+
+            if stop is None:
+                stop = start
+                start = 0
+
+            if step is None:
```

Comment:
What will happend if `arange(100,10)`?
It will described as `stop=100, step=10` in numpy

Reply:
In case of `arange(100,10)` `start=100, stop=10, step=1` as it was before. Results of `numpy` and `dpnp` are the same:
```
>>> import numpy
>>> import dpnp as np
>>> numpy.arange(100, 10)
array([], dtype=int64)
>>> print(np.arange(100, 10))
<__str__ TODO>
```

Reply:
it is really strange. Perhaps I cat not find good example for this. as far as I remember, the function `arange` is one of the special function in case of parameters handling. It is not a simply usage of `argc, kwargs`. This signature was implemented intentionally because first parameter is optional.

Are you sure this works properly and all tests non skipped for this?

Reply:
Unskipped all the tests on `arange` and they are passed. BTW the implementation is similar to `cupy` implementation https://github.com/cupy/cupy/blob/v8.1.0/cupy/_creation/ranges.py#L9.